src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,17 +21,125 @@ public abstract class InstantiateArgumentExceptionsCorrectlyFixer : CodeFixProvi
     {
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleId);
 
-        public sealed override FixAllProvider GetFixAllProvider()
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
-            return WellKnownFixAllProviders.BatchFixer;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var objectCreation = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            var diagnostic = context.Diagnostics.First();
+
+            switch (diagnostic.Properties["Problem"])
```

Comment:
There are three different things this code fixer is supposed to fix; I indicated which problem we're fixing by storing a string in the diagnostic's properties and switching on it. Do you think there's a better way to do this w/ less magic strings?

Reply:
I would create an enum and store that, converting to/from int when storing.